test/quantization/fx/test_quantize_fx.py

```diff
@@ -469,6 +470,105 @@ def forward(self, x):
         })
         self.checkGraphModuleNodes(m, expected_node=ns.call_module(MyConvReLU))
 
+    def test_fusion_turns_off_for_multi_use_intermediate_nodes(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.conv2d = nn.Conv2d(1, 1, 1)
+                self.bn2d = nn.BatchNorm2d(1)
+
+            def forward(self, x):
+                x = self.conv2d(x)
+                x = self.conv2d(x)
+                x = self.bn2d(x)
+                return x
+
+        # test train mode
+        m = M().train()
+        m = prepare_qat_fx(m, {})
+        expected_nodes = [
+            ns.call_module(nn.Conv2d),
+            ns.call_module(nn.Conv2d),
+            ns.call_module(nn.BatchNorm2d),
+        ]
+        expected_occurrence = {
+            ns.call_module(nni.ConvBn2d): 0,
+        }
+        self.checkGraphModuleNodes(
+            m,
+            expected_node_list=expected_nodes,
+            expected_node_occurrence=expected_occurrence)
+
+        # test eval mode
+        m = M().eval()
+        # fuse_fx is a top level api and only supports eval mode
+        m = fuse_fx(m)
+        expected_nodes = [
+            ns.call_module(nn.Conv2d),
+            ns.call_module(nn.Conv2d),
+            ns.call_module(nn.BatchNorm2d),
+        ]
+        expected_occurrence = {
+            ns.call_module(nni.ConvBn2d): 0,
+        }
+        self.checkGraphModuleNodes(
+            m,
+            expected_node_list=expected_nodes,
+            expected_node_occurrence=expected_occurrence)
+
+    def test_fusion_works_for_multi_use_intermediate_relu_nodes(self):
```

Comment:
there is one more test case, reported by Kimish, could you add that as well (probably need to change the matching code to address this as well), something like

```
class M(torch.nn.Module):
    def forward(self, x):
           x = self.linear(x)
           bias = self.linear.bias
           ...
```
I can take it as well if you don't have time